Velaptor/Graphics/CircleShape.cs

```diff
@@ -0,0 +1,216 @@
+        Justification = "Standard text is correct.  Reported incorrectly.")]
+    public CircleShape()
+    {
+    }
+
+    /// <summary>
+    /// Gets or sets the position of the circle.
+    /// </summary>
+    /// <remarks>
+    ///     This position is relative to the center of the circle.
+    /// </remarks>
+    public Vector2 Position { get; set; } = Vector2.Zero;
+
+    /// <summary>
+    /// Gets or sets the width of the circle.
+    /// </summary>
+    /// <remarks>
+    ///     The diameter is restricted to a minimum value of one.
+    /// </remarks>
+    public float Diameter
+    {
+        get => this.diameter;
+        set
+        {
+            value = value < 1f ? 1f : value;
+            this.diameter = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the radius of the circle.
+    /// </summary>
+    /// <remarks>
+    /// This is half of the <see cref="Diameter"/>.
+    /// </remarks>
+    public float Radius
+    {
+        get => Diameter / 2f;
+        set => Diameter = value * 2f;
+    }
+
+    /// <summary>
+    /// Gets or sets the top location of the circle.
+    /// </summary>
+    /// <remarks>
+    ///     Will automatically update the <see cref="Position"/> of the circle.
+    /// </remarks>
+    public float Top
+    {
+        get => Position.Y - Radius;
+        set => Position = new Vector2(Position.X, value + Radius);
+    }
+
+    /// <summary>
+    /// Gets or sets the right location of the circle.
+    /// </summary>
+    /// <remarks>
+    ///     Will automatically update the <see cref="Position"/> of the circle.
+    /// </remarks>
+    public float Right
+    {
+        get => Position.X + Radius;
+        set => Position = new Vector2(value - Radius, Position.Y);
+    }
+
+    /// <summary>
+    /// Gets or sets the bottom location of the circle.
+    /// </summary>
+    /// <remarks>
+    ///     Will automatically update the <see cref="Position"/> of the circle.
+    /// </remarks>
+    public float Bottom
+    {
+        get => Position.Y + Radius;
+        set => Position = new Vector2(Position.X, value - Radius);
+    }
+
+    /// <summary>
+    /// Gets or sets the Left location of the circle.
+    /// </summary>
+    /// <remarks>
+    ///     Will automatically update the <see cref="Position"/> of the circle.
+    /// </remarks>
+    public float Left
+    {
+        get => Position.X - Radius;
+        set => Position = new Vector2(value + Radius, Position.Y);
+    }
+
+    /// <summary>
+    /// Gets or sets the color of the circle.
+    /// </summary>
+    /// <remarks>
+    ///     Ignored if the <see cref="GradientType"/> is set to any value other than <see cref="ColorGradient.None"/>.
+    /// </remarks>
+    public Color Color { get; set; } = Color.White;
+
+    /// <summary>
+    /// Gets or sets a value indicating whether or not the circle is filled or empty.
+    /// </summary>
+    public bool IsFilled { get; set; } = true;
```

Comment:
Refactor name of this property to `IsSolid`.

This is to prevent confusion with the `IsEmpty()` method which is for a different context.

```suggestion
    public bool IsSolid { get; set; } = true;
```